compiler/qsc/src/interpret/circuit_tests.rs

```diff
@@ -0,0 +1,804 @@
+                }
+                for qss in q3 {
+                    for qs in qss {
+                        for q in qs {
+                            Y(q);
+                        }
+                    }
+                }
+                X(q);
+                MeasureEachZ(q1)
+            }
+        }",
+        Profile::Unrestricted,
+    );
+
+    let circ = interpreter
+        .circuit(CircuitEntryPoint::Operation("Test.Test".into()))
+        .expect("circuit generation should succeed");
+
+    expect![[r"
+        q_0    ── H ──── M ──
+                         ╘═══
+        q_1    ── H ──── M ──
+                         ╘═══
+        q_2    ── X ─────────
+        q_3    ── X ─────────
+        q_4    ── X ─────────
+        q_5    ── X ─────────
+        q_6    ── Y ─────────
+        q_7    ── Y ─────────
+        q_8    ── Y ─────────
+        q_9    ── Y ─────────
+        q_10   ── Y ─────────
+        q_11   ── Y ─────────
+        q_12   ── Y ─────────
+        q_13   ── Y ─────────
+        q_14   ── X ─────────
+    "]]
+    .assert_eq(&circ.to_string());
+}
+
+#[test]
+fn adjoint_operation() {
+    let mut interpreter = interpreter(
+        r"
+        namespace Test {
+            @EntryPoint()
+            operation Main() : Result[] { [] }
+
+            operation SWAP (q1 : Qubit, q2 : Qubit) : Unit
+                is Adj + Ctl {
+
+                body (...) {
+                    CNOT(q1, q2);
+                    CNOT(q2, q1);
+                    CNOT(q1, q2);
+                }
+
+                adjoint (...) {
+                    SWAP(q1, q2);
+                }
+
+                controlled (cs, ...) {
+                    CNOT(q1, q2);
+                    Controlled CNOT(cs, (q2, q1));
+                    CNOT(q1, q2);
+                }
+            }
+        }",
+        Profile::Unrestricted,
+    );
+
+    let circ = interpreter
+        .circuit(CircuitEntryPoint::Operation("Adjoint Test.SWAP".into()))
+        .expect("circuit generation should succeed");
+
+    expect![[r"
+        q_0    ── ● ──── X ──── ● ──
+        q_1    ── X ──── ● ──── X ──
+    "]]
+    .assert_eq(&circ.to_string());
+}
+
+#[test]
+fn lambda() {
+    let mut interpreter = interpreter(
+        r"
+        namespace Test {
+            @EntryPoint()
+            operation Main() : Result[] { [] }
+
+            operation SWAP (q1 : Qubit, q2 : Qubit) : Unit
+                is Adj + Ctl {
+
+                body (...) {
+                    CNOT(q1, q2);
+                    CNOT(q2, q1);
+                    CNOT(q1, q2);
+                }
+
+                adjoint (...) {
+                    SWAP(q1, q2);
+                }
+
+                controlled (cs, ...) {
+                    CNOT(q1, q2);
+                    Controlled CNOT(cs, (q2, q1));
+                    CNOT(q1, q2);
+                }
+            }
+        }",
+        Profile::Unrestricted,
+    );
+
+    let circ = interpreter
```

Comment:
Was this test meat to check circuit generation for `SWAP` (instead of `H`) using a lambda?

Reply:
Uh oh, copy paste error. It is actually supposed to test the lambda, but the SWAP code got left in there from a different test I copied. Thanks!